reference/multipart.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: page
+title: Multipart Encoding
+---
+
+Praxis has built-in support for handling "multipart/form-data" (link to RFC?) encoded requests and responses, in the form of the `Praxis::Types::MultipartArray` type, and the `Praxis::Responses::MultipartOk` response.
+
+
+## Praxis::Types::MultipartArray
+
+The `MultipartArray` type is an `Attributor::Collection` of `Praxis::MultipartPart` members that allows you to specify the handling of each part of a multipart request or response body.
+
+### Definition
+
+* `part 'name'`
+* `part /name/`
+* `part 'email', multiple: true`
+
+* defining parts that are files:
+  * simple: `part 'file', filename: true`
+  * simpler: `file 'file'`
+  * with filename string validation:
+    * `file 'file' { filename values: ['thumb.jpg'] }`
+    * `file 'file' { filename regex: /^img-/ }`
+  * with a filename of type other than String
+    * `file 'daily' { filename DateTime }`
+
+Example: 
+
```

Comment:
Here is a more complete (and complex) example of defining a multipart type that defines several parts in different ways for illustration purposes: